src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -46,16 +48,23 @@ public function __construct(ControllerNameParser $parser, LoggerInterface $logge
     }
 
     /**
-     * Loads a resource.
-     *
-     * @param mixed  $resource A resource
-     * @param string $type     The resource type
-     *
-     * @return RouteCollection A RouteCollection instance
+     * {@inheritdoc}
      */
     public function load($resource, $type = null)
     {
-        $collection = parent::load($resource, $type);
+        if (false !== $this->loading) {
+            // This can happen if a fatal error occurs in parent::load()
```

Comment:
The case handled here is not with an exception but with a fatal error:
- the first time the routes are loaded, a fatal error is triggered (e.g. parse error in controller while loading annotations)
- PHP abruptly empties the stack trace, bypassing all catch blocks. It then calls the registered shutdown functions.
- the ErrorHandler catched the fatal error at this stage and convert it to a FatalErrorException, then re-inject this exception to the HttpKernel with ->terminateWithException (that calls handleException())
- at this stage, if we try to load the routes again, we must prevent the fatal error from occurring a second time, that's why we have this state
- the webprofiler triggers an url generation, which requires the routes being loaded
- this patch handles the case an prevents the second fatal error by triggering an exception beforehand